tests/unit/source_code/samples/values_across_notebooks_magic_line.py

```diff
@@ -0,0 +1,4 @@
+# Databricks notebook source
+
+%run "./values_across_notebooks_child.py"
```

Comment:
is this a valid notebook code? please attach a screenshot.

Reply:
This is interesting…

It's a valid notebook:
![image](https://github.com/databrickslabs/ucx/assets/401970/8f99f5dd-191c-4634-bbb1-b2cfb0aa8631)

However there seems to be an issue[^1] at the moment where this line-magic isn't interpreted correctly in a cell that contains other code:
![image](https://github.com/databrickslabs/ucx/assets/401970/4bd9edc0-72dc-403d-92dd-1b03223beb49)
(This is limited to `%run`: things like `%pip` seem to be fine.

It works correctly if split out to a cell on its own, as the other test sample in this PR demonstrates.

[^1]: Also mentioned [here](https://community.databricks.com/t5/data-engineering/cannot-run-another-notebook-from-same-directory/td-p/68627). In Jupyter it's perfectly fine:
![image](https://github.com/databrickslabs/ucx/assets/401970/c8f10c72-0670-4c54-b4e0-cdf650a5a4e7)

Reply:
I should add that my opinion on whether we should support this is or not comes down to whether the implementation is simpler if we allow it. Given that line-magic like this is allowed in general, I can imagine that _not_ allowing the specific `%run` case might be more work than just leaving it in.

Reply:
![Screenshot 2024-07-11 at 11 16 38](https://github.com/databrickslabs/ucx/assets/3332132/58561619-6bbb-4635-a63e-c7d605f2b4e1)

Reply:
It works, but there are some limitations:
 - if `path` is a notebook, the line should be the only one in the cell (and should not have an extension)
 - if `path` is a file, it must have an extension, and the line should not be the first one
@asnare FYI

Reply:
See #2155 and #2156